scrapy/utils/log.py

```diff
@@ -232,18 +230,9 @@ def logformatter_adapter(logkws: dict) -> Tuple[int, str, dict]:
     and adapts it into a tuple of positional arguments for logger.log calls,
     handling backward compatibility as well.
     """
-    if not {"level", "msg", "args"} <= set(logkws):
-        warnings.warn("Missing keys in LogFormatter method", ScrapyDeprecationWarning)
-
-    if "format" in logkws:
-        warnings.warn(
-            "`format` key in LogFormatter methods has been "
-            "deprecated, use `msg` instead",
-            ScrapyDeprecationWarning,
-        )
 
     level = logkws.get("level", logging.INFO)
-    message = logkws.get("format", logkws.get("msg"))
+    message = str(logkws.get("msg"))
```

Comment:
Why str()? It wasn't needed before.

Reply:
Hi, wRAR,
The problem here is that `logkws.get("msg")` returns `Any | None` instead of `Any`. The previous code works well since it sets a default value in get() and returns `Any`.
If we don't take measures here, the type of variable message is `Any | None`. And an incompatible return value error arises when conducting tests.
<img width="711" alt="c0a2e8ee4fb9c2b8c01599fad954a21" src="https://github.com/scrapy/scrapy/assets/61908960/89f864b7-86fd-499b-a069-d8e49703ed83">
<img width="850" alt="f105114ba8921466f0a84ab5fa76967" src="https://github.com/scrapy/scrapy/assets/61908960/4a5cd105-1d77-4e61-a5f1-b89f167a73e0">



I came up with three possible solutions:

1. ```message = logkws.get("msg", logkws.get("msg"))```
2. ```message = logkws.get("msg", "")```
3. ```message = str(logkws.get("msg"))```

Do you have any suggestions or preferences among these three solutions?

Reply:
The usual approach here, when you can be certain that the value is a string, is to use an assert afterwards, i.e. `assert message is not None`, which should silence mypy.

Reply:
I‘ve already tested, the code below could handle both two scenarios you mentioned.
1. key `msg` is missing in the dictionary
2. Dictionary contains `"msg": None`
```python
    message = logkws.get("msg", "")
    assert message is not None
```
If there is no problem, I will modify and commit it.
@Gallaecio